inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -19,78 +18,254 @@
+            raise DispatcherException(f"No schedule manifest found with schedule_id: {schedule_id}")
+        return rows
+    
+    def _select_manifests_by_schedule_id(self, schedule_manifest_rows: list[Any]) -> list[str]:
+        # Get the manifests using the manifest_ids from the join table
+        manifests: list[str] = []
+        for row in schedule_manifest_rows:
+            # Get the manifests using the manifest_ids from the join table
+            sql = ''' SELECT * FROM manifest WHERE id = ? '''
+            try:
+                self._cursor.execute(sql, (row[2],))
+            except (sqlite3.IntegrityError, sqlite3.InternalError, sqlite3.OperationalError) as e:
+                raise DispatcherException(f"Error selecting manifest from database: {e}")
+            
+            manifest_row = self._cursor.fetchone()
+            if not manifest_row:
+                raise DispatcherException(f"No manifest found with id: {manifest_row[2]}")
+            manifests.append(manifest_row[1])
+        return manifests  
+                 
+    def _create_tables_if_not_exist(self) -> None:
+        self._create_single_schedule_table()
+        self._create_repeated_schedule_table()
+        self._create_manifest_table()
+        self._create_single_schedule_manifest_table()
+        self._create_repeated_schedule_manifest_table()
+    
+    def _create_single_schedule_table(self) -> None:      
+        sql = ''' CREATE TABLE IF NOT EXISTS single_schedule(
+                                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                                request_id TEXT NOT NULL,
+                                start_time TEXT NOT NULL,
+                                end_time TEXT); '''
+        self._conn.execute(sql)
+    
+    def _create_repeated_schedule_table(self) -> None:      
+        sql = ''' CREATE TABLE IF NOT EXISTS repeated_schedule(
+                                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                                request_id TEXT,
+                                cron_duration TEXT,
+                                cron_minutes TEXT,
+                                cron_hours TEXT,
+                                cron_day_month TEXT,
+                                cron_month TEXT,
+                                cron_day_week TEXT); '''
+        self._conn.execute(sql)
+        
+    def _create_manifest_table(self) -> None:      
+        sql = ''' CREATE TABLE IF NOT EXISTS manifest(
+                                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                                manifest TEXT); '''
+        self._conn.execute(sql)
+        
+    def _create_single_schedule_manifest_table(self) -> None:
+        sql = ''' CREATE TABLE IF NOT EXISTS single_schedule_manifest(
+                                priority INTEGER NOT NULL,
+                                schedule_id INTEGER NOT NULL,
+                                manifest_id INTEGER NOT NULL); '''
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Add explicit foreign key constraints in table creation for data integrity.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        sql = ''' CREATE TABLE IF NOT EXISTS single_schedule_manifest(
                                  priority INTEGER NOT NULL,
                                  schedule_id INTEGER NOT NULL REFERENCES single_schedule(id),
                                  manifest_id INTEGER NOT NULL REFERENCES manifest(id)); '''
```